default/python/universe_generation/galaxy.py

```diff
@@ -440,45 +746,41 @@ def calc_star_system_positions(gsd):
 
     # calculate typical width for universe based on number of systems
     width = calc_universe_width(gsd.shape, gsd.size)
-    print "Set universe width to", width
+    print "Set universe width to {}".format(width)
     fo.set_universe_width(width)
 
     positions = []
+    adjacency_grid = AdjacencyGrid(width)
 
-    print "Creating", gsd.shape, "galaxy shape"
+    print "Creating {} galaxy shape".format(gsd.shape)
     if gsd.shape == fo.galaxyShape.spiral2:
-        spiral_galaxy_calc_positions(positions, 2, gsd.size, width)
+        spiral_galaxy_calc_positions(positions, adjacency_grid, 2, gsd.size, width)
     elif gsd.shape == fo.galaxyShape.spiral3:
-        spiral_galaxy_calc_positions(positions, 3, gsd.size, width)
+        spiral_galaxy_calc_positions(positions, adjacency_grid, 3, gsd.size, width)
     elif gsd.shape == fo.galaxyShape.spiral4:
-        spiral_galaxy_calc_positions(positions, 4, gsd.size, width)
+        spiral_galaxy_calc_positions(positions, adjacency_grid, 4, gsd.size, width)
     elif gsd.shape == fo.galaxyShape.elliptical:
-        elliptical_galaxy_calc_positions(positions, gsd.size, width)
+        elliptical_galaxy_calc_positions(positions, adjacency_grid, gsd.size, width)
     elif gsd.shape == fo.galaxyShape.disc:
-        disc_galaxy_calc_positions(positions, gsd.size, width)
+        disc_galaxy_calc_positions(positions, adjacency_grid, gsd.size, width)
     elif gsd.shape == fo.galaxyShape.cluster:
-        # Typically a galaxy with 100 systems should have ~5 clusters
-        avg_clusters = gsd.size / 20
-        if avg_clusters < 2:
-            avg_clusters = 2
-        # Add a bit of random variation (+/- 20%)
-        clusters = randint((avg_clusters * 8) / 10, (avg_clusters * 12) / 10)
-        if clusters >= 2:
-            cluster_galaxy_calc_positions(positions, clusters, gsd.size, width)
+        cluster_galaxy_calc_positions(positions, adjacency_grid, gsd.size, width)
```

Comment:
Moving that piece of code into `cluster_galaxy_calc_positions` is probably a good idea. If anything, there should be a parameter specifying the numbers of stars per cluster, instead of the numbers of clusters.